packages/shared/src/child_bounties.ts

```diff
@@ -0,0 +1,2650 @@
+ */
+async function getParentTotalChildBountiesCount(client: Client<any, any>, parentIndex: number): Promise<number> {
+  return (await client.api.query.childBounties.parentTotalChildBounties(parentIndex)).toNumber()
+}
+
+/**
+ * Sets the treasury's last spend period block number to enable bounty funding
+ * @param client - The chain client
+ */
+async function setLastSpendPeriodBlockNumber(client: Client<any, any>, testConfig: TestConfig) {
+  const spendPeriod = client.api.consts.treasury.spendPeriod
+  const currentBlock = await getBlockNumber(client.api, testConfig.blockProvider)
+  const offset = schedulerOffset(testConfig)
+
+  const newLastSpendPeriodBlockNumber = currentBlock - spendPeriod.toNumber() + TREASURY_SETUP_OFFSET * offset
+  await client.dev.setStorage({
+    Treasury: {
+      lastSpendPeriod: newLastSpendPeriodBlockNumber,
+    },
+  })
+}
+
+async function extractExtrinsicFailedEvent(client: Client<any, any>): Promise<any> {
+  const events = await client.api.query.system.events()
+  const [ev] = events.filter((record) => {
+    const { event } = record
+    return event.section === 'system' && event.method === 'ExtrinsicFailed'
+  })
+
+  if (!ev) {
+    throw new Error('No ExtrinsicFailed event found')
+  }
+  return ev
+}
+
+async function scheduleInlineCallWithOriginTreasurer(
+  client: Client<any, any>,
+  encodedCall: HexString,
+  testConfig: TestConfig,
+) {
+  await scheduleInlineCallWithOrigin(client, encodedCall, { Origins: 'Treasurer' }, testConfig.blockProvider)
+}
+
+/// -------
+/// Tests
+/// -------
+
+/**
+ * Test: child bounty creation test.
+ *
+ * This test verifies the fundamental child bounty creation workflow to ensure that:
+ * - Child bounties can only be created from active parent bounties
+ * - The parent bounty curator has the authority to create child bounties
+ * - Proper events are emitted and storage is updated during creation
+ *
+ * Test structure:
+ * 1. Alice creates a parent bounty and it goes through the full lifecycle (propose → approve → fund → assign curator → accept curator)
+ * 2. Bob (as parent curator) creates a child bounty from the active parent bounty
+ * 3. Verify proper events are emitted and child bounty is stored correctly
+ */
+export async function childBountyCreationTest<
+  TCustom extends Record<string, unknown> | undefined,
+  TInitStorages extends Record<string, Record<string, any>> | undefined,
+>(chain: Chain<TCustom, TInitStorages>, testConfig: TestConfig) {
+  const [client] = await setupNetworks(chain)
+
+  await setupTestAccounts(client, ['alice', 'bob', 'charlie', 'dave'])
+
+  await setLastSpendPeriodBlockNumber(client, testConfig)
+
+  await client.dev.newBlock()
+
+  const bountyValueMinimum = client.api.consts.bounties.bountyValueMinimum
+  const bountyValue = bountyValueMinimum.toBigInt() * BOUNTY_MULTIPLIER // 1000 tokens
+  const description = 'Test bounty for child bounty creation'
+
+  // propose a bounty
```

Comment:
Could you add some comments to these longer tests that delineate where each section ends/begins?

I.e. in this test's top-level comment, you have a `Test structure:` section; inside the test, you could add comments that specific where `1/2/3` start.